### DISCUSSION TITLE
Actions started to fail after MinGW was updated from 11.2.0 to 12.2.0

### DISCUSSION THREAD

**Discussion by TheMostDiligent** [actions/runner#2877] [Q&A]
Hello!

After Windows runner was recently updated to use MinGW 12.2.0, we started seeing [the following error](https://github.com/DiligentGraphics/DiligentEngine/actions/runs/6269903295/job/17027039561#step:7:24) when running tests:

```
Error: Process completed with exit code 127.
```

Nothing has changed and the tests that used [to run fine](https://github.com/DiligentGraphics/DiligentEngine/actions/runs/6250356280/job/16969073831#step:7:23) with [11.2.0](https://github.com/DiligentGraphics/DiligentEngine/actions/runs/6250356280/job/16969073831#step:5:31) now fail to start. Locally, everything works fine with both versions. I am very confused by this problem, so any suggestion would be highly appreciated.

**Comment by bryanmacfarlane** [comment]
> D:/a/DiligentEngine/build/DiligentCore/Tests/DiligentCoreTest/DiligentCoreTest.exe

It looks like this process is failing so it seems external to the runner.  the runner is just running it and reporting the return code.  Have you tried to get a windows vm with mingw 12.2.0 on it to debug it?  not sure if ya'll created that exe or not.  if not, then maybe follow up with the author?

**Reply by TheMostDiligent** [reply]
The only thing that differs between the successful run and the failing run is the runner version. Our build setup is identical in both cases.
I have tried mingw 12.2.0 on Windows and everything works.
About 12.2.0, it would be nice if there was some clarification which flavor of mingw is used on the runner (there is a lot of them).

**Reply by ChristopherHX** [reply]
The runner image changed, the runner itself doesn't change that much see <https://github.com/actions/runner-images/releases/tag/win22%2F20230918.1>

You find there a script to recreate the GitHub Hosted Runner Image in an azure vm (you need to understand the instructions..., except some files irrelevant to your problem) you can selfhost for testing.

In the azure vm you could try to debug it as you wish, should match the software stack with the problem you are facing.

**Reply by TheMostDiligent** [reply]
I actually suspect there might be some issue with the mingw on the image.
Can you please provide information which mingw package you are using?
There are [quite a lot of options](https://winlibs.com/).

**Reply by ChristopherHX** [reply]
I'm not using mingw nor I'm a member of GitHub. 

It's mingw from chocolatey https://github.com/actions/runner-images/blob/af8120a33b448350ba4d4a7d7e919b38b09a86d2/images/win/scripts/Installers/Install-Mingw64.ps1#L7

I linked you where you find all install scripts, they are published by GitHub

**Reply by bryanmacfarlane** [reply]
I wonder if this issue should be moved to the runner-images repo (if the issue is with mingw).  e.g. should they rollback to the previous version etc?  is it broken or is it issue between tis exe and 12.2 etc?

**Reply by TheMostDiligent** [reply]
Yes, that sounds right, but I can't move this discussion as I don't have rights.
